docs/development.rst

```diff
@@ -133,6 +133,137 @@ To get an idea about the size distribution of our react codebase you can run::
+
+    from kuma.example.views import views
+    ....
+    urlpatterns += i18n_patterns(
+        re_path(r"example/?$", views.example, name="example"),
+    )
+
+#. We are done with the Django part! In your browser, go to ``/example`` and you should see a blank page that says "This works!"
+
+#. Now we can set up the React part. In ``example.html``, we can remove the "This works!" text and add the following code between ``{% block title %}`` and ``{% block content %}``::
+
+    {# Pass `True` for initial data if you do not have any data to fetch. Otherwise, change `True` to `None` or whatever your initial data will be. #}
+    {% block document_head %}
+        {{ render_react("SPA", request.LANGUAGE_CODE, request.get_full_path(), True)|safe }}
+    {% endblock %}
+
+#. So your ``example.html`` file should now look something like this::
+
+    {% extends "react_base.html" %}
+
+    {% block title %}{{ page_title(_('My Page Title')) }}{% endblock %}
+
+    {% block document_head %}
+        {{ render_react("SPA", request.LANGUAGE_CODE, request.get_full_path(), True)|safe }}
+    {% endblock %}
+
+    {% block content %}{% endblock %}
+
+    {% block footer %}{% endblock %}
+
+
+#. In ``kuma/javascript/src``, we need to create our React component, ``example.jsx``. Add the following lines as a starter::
+
+    // @flow
+    import * as React from 'react';
+
+    import { gettext } from './l10n.js';
+    import A11yNav from './a11y/a11y-nav.jsx';
+    import Header from './header/header.jsx';
+    import Footer from './footer.jsx';
+    import Route from './route.js';
+
+    type ExampleRouteParams = {
+        locale: string
+    };
+
+    export default function ExamplePage() {
+        return (
+            <>
+                <A11yNav />
+                <Header />
+                <Footer />
+            </>
+        );
+    }
+
+    // In order to use new URL() with relative URLs, we need an absolute base
+    // URL. If we're running in the browser we can use our current page URL.
+    // But if we're doing SSR, we just have to make something up.
+    const BASEURL =
+        typeof window !== 'undefined' && window.location
+            ? window.location.origin
+            : 'http://ssr.hack';
+
+    export class ExampleRoute extends Route<ExampleRouteParams, null> {
+        locale: string;
+
+        constructor(locale: string) {
+            super();
+            this.locale = locale;
+        }
+
+        getComponent() {
+            return ExamplePage;
+        }
+
+        match(url: string): ?ExampleRouteParams {
+            const path = new URL(url, BASEURL).pathname;
+            const examplePath = `/${this.locale}/example`;
+            const regex = new RegExp(examplePath, 'g');
+
+            if (regex.test(path)) {
+                return {
+                    locale: this.locale
+                };
+            }
+            return null;
+        }
+
+        fetch() {
+            return Promise.resolve(null);
```

Comment:
Perhaps we should add a note about the fact that this component uses static data and that is why this instance of the `fetch` function retuns `null`. For now, we can then just point to an example in the codebase of a component that does actually need initial data.

At a later stage we can then add another example of a component that does implement `fetch`.

Reply:
I ended up going down a rabbit hole trying to understand the Router component! I ended up adding examples for `fetch` and initial data. It turns out static pages do not need the `fetch()` method at all.